client/src/types.ts

```diff
@@ -1,18 +1,51 @@
+import * as Immutable from 'immutable';
+
+const shallowClone = (a: any[]) => [...a];
+
 export interface Author {
   personId: string;
   name: string;
 }
 
-export interface Bit {
+export type Vote = -1 | 0 | 1;
+
+export class Bit {
```

Comment:
Is this a common TS pattern? Should we really shove all our data types in a single file?

Reply:
No we should group them more topically.  Like we should have a file for bits (and related types), a file for users (and related types), etc. i just named it types for now because we have no types yet and I wanted somewhere to start putting anything.